content/automate/ManageIQ/Service/Retirement/StateMachines/Methods.class/__methods__/check_service_retired.rb

```diff
@@ -1,37 +1,71 @@
-#
-# Description: This method checks to see that all of the service resources are retired before retiring the service.
-#
-
-service = $evm.root['service']
-if service.nil?
-  $evm.log('error', "Service Object not found")
-  exit MIQ_ABORT
-end
+module ManageIQ
+  module Automate
+    module Service
+      module Retirement
+        module StateMachines
+          module Methods
+            class CheckServiceRetired
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                service
```

Comment:
@d-m-u Could we set a local service attribute here so we can pass it into the check_service_resource_retirement and result methods?